File: src/codes/no-undo-history/ts/ksrc.ts
class DrawService {
  private canvas: string[] = [];

  draw(shape: string) {
    this.canvas.push(shape);
    console.log(`描画: ${shape}`);
  }

  showCanvas() {
    console.log("現在のキャンバス:", this.canvas.join(", "));
  }
}

// 利用例
const drawer = new DrawService();
drawer.draw("circle");
drawer.draw("square");
drawer.showCanvas(); // circle, square

File: src/codes/no-undo-history/py/ksrc.py
class DrawService:
    def __init__(self):
        self.canvas = []

    def draw(self, shape: str):
        self.canvas.append(shape)
        print(f"描画: {shape}")

    def show_canvas(self):
        print("現在のキャンバス:", ", ".join(self.canvas))


# 利用例
drawer = DrawService()
drawer.draw("circle")
drawer.draw("square")
drawer.show_canvas()  # circle, square
